### DISCUSSION TITLE
Quarto is using JuliaCall/R to execute Julia. Can I control environment? JuliaCall still good to run all Julia?

### DISCUSSION THREAD

**Discussion by ghost** [quarto-dev/quarto-cli#8236] [Q&A]
### Description

It looks like Quarto is using JuliaCall/R to execute Julia code. I have Julia, Python, and R in the doc. Can I control environment, like using the environment in a folder like "Analysis1" that as a Manifest.toml and Project.toml instead of the base environment (v1.9)? Also, is JuliaCall still good to run all Julia code normally? It works for this basic test, but I plan on doing much more complicated machine learning stuff.

Here's an example of the code that I'm testing:
````
---
title: "JPR Test"
---

```{python}
#| label: fig-polar
#| fig-cap: "A line plot on a polar axis"

import numpy as np
import matplotlib.pyplot as plt

r = np.arange(0, 2, 0.01)
theta = 2 * np.pi * r
fig, ax = plt.subplots(
  subplot_kw = {'projection': 'polar'} 
)
ax.plot(theta, r)
ax.set_rticks([0.5, 1, 1.5, 2])
ax.grid(True)
plt.show()
```

```{julia}
#| label: fig-plots
#| fig-cap: "Multiple Plots"
#| fig-subcap:
#|   - "Plot 1"
#|   - "Plot 2"
#| layout-ncol: 2

using Plots
display(plot(sin, x -> sin(2x), 0, 2))
display(plot(x -> sin(4x), y -> sin(5y), 0, 2))
```

```{r}
#| label: fig-airquality
#| fig-cap: "Temperature and ozone level."
#| warning: false

library(ggplot2)

ggplot(airquality, aes(Temp, Ozone)) + 
  geom_point() + 
  geom_smooth(method = "loess")
```
````

**Comment by mcanouil** [comment]
Please format your code properly using more backticks than you have in your actual code, for instance four backticks.

This being said, if you are using R code cells, knitr will be chosen as the engine, otherwise Jupyter.
Use `engine` to explicitly ask an engine as documented (<https://quarto.org/docs/computations/execution-options.html#engine-binding>).

Note that (also documented), you cannot mix Jupyter AND knitr.
So if you choose to use Jupyter, then you'll need to add the Python module or Julia library required to run R code through Python or Julia in your case.

You are using the latest stable which has patches for "important" bugs, so you should upgrade to the latest stable version if not on 1.4 release candidate.

**Comment by cderv** [comment]
> It looks like Quarto is using JuliaCall/R to execute Julia code. I have Julia, Python, and R in the doc.

Quarto will either use Jupyter engine or Knitr engine (https://quarto.org/docs/computations/execution-options.html#engine-binding). There is a automatic engine binding based on document content. 

Only knitr engine supports having different language cell in the same document. Jupyter will only work with a single Jupyter Kernel. 

So 

- A quarto doc with only Julia code will by default be run using IJulia through Jupyter
- A quarto doc with R + Python + Julia will use knitr and Python cell will be run with [reticulate](https://rstudio.github.io/reticulate/) while Julia cell will be run with [JuliaCall](https://non-contradiction.github.io/JuliaCall/index.html). Both are R packages that allows to compute in another language and have knitr chunk support. 

> Can I control environment, like using the environment in a folder like "Analysis1" that as a Manifest.toml and Project.toml instead of the base environment (v1.9)

Considering which tools run Julia, you can now see that the configuration will be done in R for JuliaCall. You should get through their documentation and repo . Quick search lead me to 

* https://github.com/Non-Contradiction/JuliaCall/issues/218

and [JULIA_PROJECT](https://docs.julialang.org/en/v1/manual/environment-variables/#JULIA_PROJECT) environment variable

There could be other support or way. 

Hope it helps
